### ISSUE TITLE
References not working for field names with dashes

### ISSUE THREAD

**Issue by santospro** [jquense/yup#156]
This is the schema I have:
```
const schema = yup.object({
	'first-field': yup.string().min(3),
	'second-field': yup.string().when(
    'first-field', (firstValue, schema) =>
    	schema.test(
        'custom-check', '', secondValue =>
        firstValue === 'token' && ['a', 'b', 'c'].includes(secondValue)
      )
  )
});
```
This is how I'm validating the schema:
```
schema.validate({
	'first-field': 'token',  
	'second-field': 'a'
}, {
	abortEarly: false
}).then(function() {
	console.log('ok', arguments);
})
.catch(function(error) {
	console.log('failed', error.inner);
})
```

And I'm getting the following error:

> Uncaught ReferenceError: "field" is not defined
>     at Reference.eval [as _get] (eval at getter (node_modules/property-expr/index.js:27), <anonymous>:3:20)
>     at Reference.getValue (lib/Reference.js:55)

The error seems to be in the Expr repository: https://github.com/jquense/expr/issues/2

**Comment by corydeppen** [comment]
This may be resolved by jquense/expr#4. I did `yarn upgrade yup` in my project and the yup dependencies, including property-expr, were updated.

**Comment by zavirvla** [comment]
Hi,

This problem wasn't resolved (or was resolved partially). I have discovered "ReferenceError: integratedSecurity is not defined" during using of dashes in names within "when" conditions.
I replaced "-" with "_" in names and now everything works.
So mssql-integratedSecurity to mssql_integratedSecurity fixed the problem.
Yup version is 0.28.3

Case when error occures:
```javascript
validationSchema={yup.object().shape({
    mssql-integratedSecurity: yup.boolean(),
    mssql-username: yup
      .mixed
      .when("mssql-integratedSecurity", {
        is: true,
        then: yup.string().required()
      })
  })
}
```

Looks like the error occurs only with "when" because until this moment everything worked fine
